overlord/devicestate/firstboot.go

```diff
@@ -238,6 +243,14 @@ func populateStateFromSeedImpl(st *state.State, tm timings.Measurer) ([]*state.T
 		}
 		infos = append(infos, info)
 		infoToTs[info] = ts
+		allSnapInfos[info.InstanceName()] = info
+	}
+
+	// validate that all snaps have bases
+	errs := image.CheckBasesAndProviders(allSnapInfos)
+	if errs != nil {
+		// only report the first error encountered
+		return nil, errs[0]
```

Comment:
I wonder if we should do something similar to https://github.com/snapcore/snapd/pull/7219/files#diff-da07f373046a0a24e67b18699dc15a70L845 here, i.e. build an error that contains all the individual issues? Maybe we can make the result from "CheckBaseAndProviders" something like: 
```go
type MultiErr struct {
  Header string
  errs []errors
}
func (e MultiErr) Add(errs []error) { ... }
func (e MultiErr) Error() string {
  var buf bytes.Buffer
  for _, err := range e.errs {
	fmt.Fprintf(&buf, "\n- %s", err)
   }
   return fmt.Sprintf("%s:%s", e.Header, buf.Bytes())
}
```
but looking at it it (after I typed it out) it seems like this should probably a separate PR and get a :+1: from @pedrons first :)

Reply:
Right, I was considering this and I'm happy to reconsider. The reason I did it this way is `prepare-image` validates all aspects of the image and reports all errors at once and this is useful summary for the user when assembling an image. However, when the image is already booted I thought it's fine to report just one error which is in-line with error reporting we normally do in snapd.

Reply:
yes, I don't think it's worth the complexity in seeding code itself.

Reply:
I misread the issue, I'm fine either way given this will end up in snap/ code.